crates/telio-nurse/src/aggregator.rs

```diff
@@ -327,11 +327,19 @@ impl ConnectivityDataAggregator {
         }
     }
 
-    #[allow(dead_code)]
-    pub(crate) async fn collect_unacknowledged_segments(
-        &self,
-        force_close: bool,
-    ) -> AggregatorCollectedSegments {
+    /// Clear current peer and relay monitoring segments
+    pub async fn clear_ongoinging_segments(&self) {
+        let mut data_guard = self.data.lock().await;
+        data_guard.current_peer_events.clear();
+        data_guard.current_relay_event.take();
+    }
+
+    /// Saves unacknowledged segments in a buffer
+    ///
+    /// # Arguments
+    ///
+    /// * `force_save` - Whether to force save current events
+    pub async fn save_unacknowledged_segments(&self, force_save: bool) {
```

Comment:
Imho it would be better to make it private and create a public function without arguments